src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/BulkPrintRequestTest.java

```diff
@@ -0,0 +1,108 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@TestPropertySource(
+    properties = {
+        "document_management.api_gateway.url=false",
+        "core_case_data.api.url=false",
+        "send-letter.url=http://localhost:${wiremock.server.port}/send"
+    }
+)
+@AutoConfigureWireMock(port = 0)
+public class BulkPrintRequestTest extends BaseSaveTest {
+
+    @Autowired
+    WireMockServer wireMockServer;
+
+    @Value("${send-letter.url}")
+    private String url;
+
+    @MockBean
+    private BulkPrintStaffNotificationService bulkPrintNotificationService;
+
+    @Test
+    public void shouldNotSendNotificationWhenEverythingIsOk() throws Exception {
+        when(authTokenGenerator.generate()).thenReturn(AUTHORISATION_TOKEN);
+
+        wireMockServer.stubFor(post(urlEqualTo("/send/letters"))
+            .willReturn(aResponse()
+                .withStatus(HttpStatus.OK.value())
+                .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .withBody("{ \"letter_id\":\"" + UUID.randomUUID().toString() + "\" }")
+            )
+        );
+
+        MvcResult result = makeRequest(SampleClaimData.submittedByClaimant())
+            .andExpect(status().isOk())
+            .andReturn();
+
+        verify(bulkPrintNotificationService, never())
+            .notifyFailedBulkPrint(any(List.class), eq(deserializeObjectFrom(result, Claim.class)));
+    }
+
+    @Test
+    public void shouldTrySendingLetterThreeTimesOnFailuresWhenBulkPrintFailsWithSendLetterException() throws Exception {
+        when(authTokenGenerator.generate()).thenReturn(AUTHORISATION_TOKEN);
+
+        wireMockServer.stubFor(post(urlEqualTo("/send/letters"))
+            .willReturn(aResponse()
+                .withStatus(HttpStatus.BAD_REQUEST.value())
+                .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            )
+        );
+
+        MvcResult result = makeRequest(SampleClaimData.submittedByClaimant())
+            .andExpect(status().isOk())
+            .andReturn();
+
+        verify(bulkPrintNotificationService)
+            .notifyFailedBulkPrint(any(List.class), eq(deserializeObjectFrom(result, Claim.class)));
+    }
+
+    @Test
+    public void shouldTrySendingLetterThreeTimesOnFailuresWhenBulkPrintFailsWithRestClientException() throws Exception {
+
+        when(authTokenGenerator.generate()).thenReturn(AUTHORISATION_TOKEN);
+
+        wireMockServer.stubFor(post(urlEqualTo("/send/letters"))
+            .willReturn(aResponse()
+                .withStatus(HttpStatus.INTERNAL_SERVER_ERROR.value())
```

Comment:
Can you explain the difference in the way application handles 400 response and 500 response? I am trying to see why we have two test cases simulating these two response code.

Reply:
it has no purpose as it is handling both exceptions in same way. However wanted to keep two exceptions in sendLetterApi, if other user want to handle them separately.